csharp/test/Ice/ami/AllTests.cs

```diff
@@ -134,3188 +124,420 @@ public virtual void called()
                 private bool _called;
             }
 
-            private class AsyncCallback : CallbackBase
+            private class SentCallback : CallbackBase
             {
-                public AsyncCallback()
-                {
-                }
-
-                public AsyncCallback(Cookie cookie)
+                public SentCallback()
                 {
-                    _cookie = cookie;
+                    _thread = Thread.CurrentThread;
                 }
 
                 public void
-                isA(Ice.AsyncResult result)
+                sent(bool ss)
                 {
-                    test(result.AsyncState == _cookie);
-                    test(result.getProxy().end_ice_isA(result));
-                    called();
-                }
+                    test(ss && _thread == Thread.CurrentThread || !ss && _thread != Thread.CurrentThread);
```

Comment:
This logic doesn't work properly with await. I didn't try to change it in this PR.